lxd/device/nic_bridged.go

```diff
@@ -197,6 +197,63 @@ func (d *nicBridged) validateConfig(instConf instance.ConfigReader) error {
 		}
 	}
 
+	// Check there isn't another NIC with the any of the same addresses specified on the same cluster member.
+	// Can only validate this when the instance is supplied (and not doing profile validation).
+	if d.inst != nil && (d.config["ipv4.address"] != "" || d.config["ipv6.address"] != "") {
+		err := d.state.Cluster.InstanceList(nil, func(inst db.Instance, p api.Project, profiles []api.Profile) error {
```

Comment:
Don't we have a version of this which filters at the SQL level for instances that are local to the node?

If we're on a cluster with tens of thousand of instances, having to pull all the data for every device is going to get quite expensive. If we could do the node and project filter at the SQL level, that'd improve speed quite a lot.

Reply:
Actually, possibly a stupid question there, but wouldn't it be a LOT faster to parse the dnsmasq hosts file for that network?

Reply:
Anyway, merging the current branch as the fix is correct and it will resolve those two issues, just thinking about ways to get the same result faster and with less DB data transfer.

Reply:
@stgraber Yes, in fact the `InstanceList` fucntion takes a filter argument that is passed to `GetInstances` and is applied at the SQL level. We use that in `SRIOVGetHostDevicesInUse` already, so I'll update this to use the same approach.

As for parsing the dnsmasq hosts file, I wanted to avoid that for 2 reason:

1. I try and avoid depending on external config files, especially as we are potentially looking at replacing some of dnsmasqs usage in the future.
2. This wouldn't be available when checking unmanaged networks or networks with DHCP disabled.

But by using the filtered `InstanceList` function I think we can get some significant improvements in the scenario you mentioned above.

Reply:
@stgraber I do think we could improve/extend the `InstanceList` function (or a new function) in the future that allows pre-filtering of certain device types (perhaps `DeviceList` function that returns instances that have certain devices), as that is a common use of the `InstanceList` function.

Reply:
https://github.com/lxc/lxd/pull/8982